db/offchaindb/db.go

```diff
@@ -0,0 +1,483 @@
+	InvalidVoteStatus VoteStatusType = iota
+
+	// NotAuthorizedVoteStatus defines status where the vote has not been
+	// authorized by author
+	NotAuthorizedVoteStatus
+
+	// AuthorizedVoteStatus defines status where the vote has been authorized by author.
+	AuthorizedVoteStatus
+
+	// StartedVoteStatus defines started votes status.
+	StartedVoteStatus
+
+	// FinishedVoteStatus defines the finished votes status.
+	FinishedVoteStatus
+
+	// NotFoundVoteStatus defines the not found votes status(doesn't exist).
+	NotFoundVoteStatus
+	UnknownVoteStatus
+)
+
+func (s VoteStatusType) String() string {
+	switch s {
+	case InvalidVoteStatus:
+		return "invalid"
+	case NotAuthorizedVoteStatus:
+		return "not authorized"
+	case AuthorizedVoteStatus:
+		return "authorized"
+	case StartedVoteStatus:
+		return "started"
+	case FinishedVoteStatus:
+		return "finished"
+	case NotFoundVoteStatus:
+		return "doesn't exit"
+	default:
+		return "unknown"
+	}
+}
+
+// VoteStatusTypeFromStr string version of the status to VoteStatusType.
+func VoteStatusTypeFromStr(val string) VoteStatusType {
+	switch val {
+	case "invalid":
+		return InvalidVoteStatus
+	case "not authorized":
+		return NotAuthorizedVoteStatus
+	case "authorized":
+		return AuthorizedVoteStatus
+	case "started":
+		return StartedVoteStatus
+	case "finished":
+		return FinishedVoteStatus
+	case "doesn't exit":
+		return NotFoundVoteStatus
+	default:
+		return UnknownVoteStatus
+	}
+}
+
+// ProposalStatusType defines the various
+type ProposalStatusType int8
+
+const (
+	// InvalidProposalStatus defines the Invalid proposal status
+	InvalidProposalStatus ProposalStatusType = iota
+
+	// NotFoundProposalStatus defines not found proposal status
+	NotFoundProposalStatus
+
+	// NotReviewedProposalStatus defines a proposal that has not been reviewed.
+	NotReviewedProposalStatus
+
+	// CensoredProposalStatus defines a proposal that has been censored.
+	CensoredProposalStatus
+
+	// PublicProposalStatus defines a  proposal that is publicly visible.
+	PublicProposalStatus
+
+	// UnreviewedChangesProposalStatus defines a proposal that is not public and
+	// has unreviewed changes.
+	UnreviewedChangesProposalStatus
+
+	// AbandonedProposalStatus defines a proposal that has been declared
+	// abandoned by an admin.
+	AbandonedProposalStatus
+	UnknownProposalStatus
+)
+
+func (p ProposalStatusType) String() string {
+	switch p {
+	case InvalidProposalStatus:
+		return "invalid"
+	case NotFoundProposalStatus:
+		return "not found"
+	case NotReviewedProposalStatus:
+		return "not reviewed"
+	case CensoredProposalStatus:
+		return "censored"
+	case PublicProposalStatus:
+		return "public"
+	case UnreviewedChangesProposalStatus:
+		return "unreviewed changes"
+	case AbandonedProposalStatus:
+		return "abandoned"
+	default:
+		return "unknown"
+	}
+}
+
+// ProposalStatusFrmStr converts the string into ProposalStatusType value.
+func ProposalStatusFrmStr(val string) ProposalStatusType {
```

Comment:
All this stuff is in `piapi` too. `PropStatus`, etc.

 https://github.com/decred/politeia/blob/73ba15aa6d3e8ba590f5cf5fc1120938997283d6/politeiawww/api/v1/v1.go#L321